go/vt/discovery/endpointwatcher_test.go

```diff
@@ -0,0 +1,136 @@
+package discovery
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	pbt "github.com/youtube/vitess/go/vt/proto/topodata"
+	"github.com/youtube/vitess/go/vt/topo"
+	"github.com/youtube/vitess/go/vt/topo/test/faketopo"
+	"golang.org/x/net/context"
+)
+
+func TestEndPointWatcher(t *testing.T) {
+	ft := newFakeTopo()
+	fhc := newFakeHealthCheck()
+	t.Logf(`ft = FakeTopo(); fhc = FakeHealthCheck()`)
+	epw := NewEndPointWatcher(topo.Server{Impl: ft}, fhc, []string{"aa"}, 10*time.Millisecond)
+	t.Logf(`epw = EndPointWatcher(topo.Server{ft}, fhc, ["aa"], 10ms)`)
+
+	ft.AddTablet("aa", 0, "host1", map[string]int32{"vt": 123})
+	time.Sleep(100 * time.Millisecond)
+	t.Logf(`ft.AddTablet("aa", 0, "host1", {"vt": 123}); Sleep(100ms)`)
+	want := &pbt.EndPoint{
+		Uid:     0,
+		Host:    "host1",
+		PortMap: map[string]int32{"vt": 123},
+	}
+	key := endPointToMapKey(want)
+	if ep, ok := fhc.endPoints[key]; !ok || len(fhc.endPoints) != 1 {
+		t.Errorf("fhc.endPoints[key] = %+v; want %+v", ep, want)
+	}
+
+	ft.AddTablet("aa", 0, "host1", map[string]int32{"vt": 456})
+	time.Sleep(100 * time.Millisecond)
+	t.Logf(`ft.AddTablet("aa", 0, "host1", {"vt": 456}); Sleep(100ms)`)
+	want = &pbt.EndPoint{
+		Uid:     0,
+		Host:    "host1",
+		PortMap: map[string]int32{"vt": 456},
+	}
+	key = endPointToMapKey(want)
+	if ep, ok := fhc.endPoints[key]; !ok || len(fhc.endPoints) != 1 {
+		t.Errorf("fhc.endPoints[key] = %+v; want %+v", ep, want)
+	}
+
+	epw.Stop()
+}
+
+func newFakeHealthCheck() *fakeHealthCheck {
```

Comment:
I would move this class to its own test file, as it's gonna be used by other tests too. Maybe fake_health_check_test.go.